src/textual/css/stylesheet.py

```diff
@@ -122,6 +122,8 @@ def __init__(self, *, variables: dict[str, str] | None = None) -> None:
         self._rules: list[RuleSet] = []
         self.variables = variables or {}
         self.source: dict[str, str] = {}
+        # Records which of the source keys represent CSS defined at the widget level
+        self._widget_css_paths: set[str] = set()
```

Comment:
I was actually on the fence about doing something like this but opted against it as it made the lookups into source a bit uglier and there were a good few of them. It's a trade-off though - happy to make the change though if you prefer it.

Reply:
Let’s give it a shot 👍

Reply:
No objections to a NamedTuple instead?

Reply:
I try not to use too many namedtuples and dataclasses, because they do add a slight penalty to startup. But if you find it makes the code more readable, go for it...